dv/verilator/pcount/cpp/ibex_pcounts.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <numeric>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "ibex_pcounts.h"
+
+// see mhpmcounter_incr signals in rtl/ibex_cs_registers.sv for details
+
+const std::vector<std::string> ibex_pcount_names = {
+    "Cycles",
+    "NONE",
+    "Instructions Retired",
+    "LSU Busy",
+    "Fetch Wait",
+    "Loads",
+    "Stores",
+    "Jumps",
+    "Conditional Branches",
+    "Taken Conditional Branches",
+    "Compressed Instructions"};
+
+std::string ibex_pcount_string(uint64_t pcounts[], bool csv) {
+  char seperator = csv ? ',' : ':';
+  std::string::size_type longest_name_length;
+
+  if (!csv) {
+    longest_name_length = std::accumulate(
```

Comment:
Don't you just want a loop like `for (const std::string &pcount_name : ibex_pcount_names) { /* logic here to update cur_longest */ }` to find the longest? (Or using auto if preferred - the codebase I normally work on limits use of auto quite a bit).

Reply:
Yes, I like the std algorithms in general, but in this case it obfuscates it whereas this is shorter+clearer.
```
for (const std::string &s : ibex_pcount_names) {
  cur_longest = std::max(cur_longest, s.length());
}
```